src/common/snippets/src/pass/fq_decomposition.cpp

```diff
@@ -205,48 +207,100 @@ bool ngraph::snippets::pass::FakeQuantizeDecomposition::getScalesAndShifts(
     if (!input_low_constant || !input_high_constant || !output_low_constant || !output_high_constant)
         return false;
 
+    const auto input_low_pshape = input_low_constant->get_output_partial_shape(0);
+    const auto input_high_pshape = input_high_constant->get_output_partial_shape(0);
+    const auto output_low_pshape = output_low_constant->get_output_partial_shape(0);
+    const auto output_high_pshape = output_high_constant->get_output_partial_shape(0);
+
     auto input_low = input_low_constant->cast_vector<float>();
     auto input_high = input_high_constant->cast_vector<float>();
     auto output_low = output_low_constant->cast_vector<float>();
     auto output_high = output_high_constant->cast_vector<float>();
     auto levels = fq_node->get_levels();
 
-    const auto input_size = std::max(input_low.size(), input_high.size());
-    const auto output_size = std::max(output_low.size(), output_high.size());
-
-    cl = input_low;
-    ch = input_high;
-    isc.resize(input_size, 0);
-    ish.resize(input_size, 0);
-    osc.resize(output_size, 0);
-    osh.resize(output_size, 0);
-
-    for (size_t i = 0; i < input_size; i++) {
-        float il = input_low[input_low.size() == 1 ? 0 : i];
-        float ih = input_high[input_high.size() == 1 ? 0 : i];
-
-        isc[i] = (levels - 1) / (ih - il);
-        ish[i] = -il * isc[i];
+    // Calculations of input scales and shift:
+    //   - isc := (levels-1) / (ih - il)
+    //   - ish := -il * isc
+    if (input_low_pshape == input_high_pshape || shape_size(input_low_pshape.get_shape()) == 1 || shape_size(input_high_pshape.get_shape()) == 1) {
+        const auto input_size = std::max(input_low.size(), input_high.size());
+        isc.resize(input_size, 0);
+        ish.resize(input_size, 0);
+        for (size_t i = 0; i < input_size; i++) {
+            float il = input_low[input_low.size() == 1 ? 0 : i];
+            float ih = input_high[input_high.size() == 1 ? 0 : i];
+
+            isc[i] = (levels - 1) / (ih - il);
+            ish[i] = -il * isc[i];
+        }
+        cl = input_low;
+        ch = input_high;
+    } else {  // NUMPY broadcasting
```

Comment:
looks like semantically we can always use NUMPY broadcasting. Why do we leave legacy computation logic for some specific cases?
The same question about output scales/shifts computation

Reply:
Because `ngraph::runtime::reference::autobroadcast_binop()` is expensive call. I'd like to not increase model compilation time.
Moreover, the case "NUMPY broadcasting" is rare in real models. I mean the situation when weights of FQ are scalar or have equal shapes is base - I haven't seen models with first case, only in test cases. So for that case we have optimal calculations. For other rare cases - common computations. Thank you!

Reply:
Ok, thanks for the explanation. Let's leave this logic as is.

Reply:
Added in 487198f95d917a0cddf520341a60b6ebfd2b687d